### ISSUE TITLE
Use oauth to save private likes or blogs marked dashboard-only

### ISSUE THREAD

**Issue by nightpool** [bbolli/tumblr-utils#128]
Many of my blogs, and many of those i follow, enabled what is known as the "dashboard-only" option, which disables the `https://username.tumblr.com` link and (more importantly, for this issue) disables unauthenticated API access. Some blogs (those marked sensitive under the previous regime) have had this state forced on them. Additionally, many users don't make their likes public, and would like to back them up anyway.

Being able to specify an access_token for requests would make it possible to back up these blogs. Having a tumblr app and oauth flow built into the script is not required, but may be a nice improvement, once the basics have been implemented.

**Comment by MrEldritch** [comment]
Agreed - this is a _critical_ feature, especially as most of the blogs that would be in need of urgent archival are NSFW blogs. Right now TumblThree is the only archiver I know of that offers this, but TumblThree is primarily an image ripper and lacks the ability to download a complete blog in readable format.

**Comment by Cebtenzzre** [comment]
[This comment](https://github.com/johanneszab/TumblThree/issues/222#issuecomment-382501234) sheds some light on how it is done in TumblThree.

**Comment by Hrxn** [comment]
I should probably mention this here, could be of interest for @Cebtenzzre and others:

https://github.com/mikf/gallery-dl

CLI, cross-platform, written in Python as well (3) and supports Tumblr, among many others.
The point is, it supports OAuth. Sans external dependencies (only `requests` for Python).
So for anyone well versed enough in Python and willing to give it shot, here are probably good starting points:
https://github.com/mikf/gallery-dl/blob/master/gallery_dl/oauth.py
https://github.com/mikf/gallery-dl/blob/master/gallery_dl/extractor/tumblr.py

I've used this (still am, actually) myself, and it worked for the kind of *special* blogs you mention here.

But please note, this was never about doing a full Tumblr backup unlike this project here.
That means that text posts will probably not work. By design. Definitely not out of the box, you have to change this config setting here: https://github.com/mikf/gallery-dl/blob/master/docs/configuration.rst#extractortumblrposts

(Which, to be fair, you have to do anyway, when setting your personal tokens for OAuth in the config file)

I can only say that I've never done any text posts so far.

But if someone is willing I'm sure that the OAuth implementation can be adapted from there. Or, alternatively, send over a PR to enhance support for other post types.
Whatever you prefer..
